java-data/src/main/java/net/cryptic_game/backend/data/user/User.java

```diff
@@ -102,12 +102,13 @@ public boolean verifyPassword(final String input) {
     }
 
     /**
+     * @param session the sql session
      * Deletes the {@link User}.
      */
     @Override
-    public void delete() {
-        net.cryptic_game.backend.data.user.Session.getByUser(this).forEach(TableModel::delete);
-        super.delete();
+    public void delete(final Session session) {
+        net.cryptic_game.backend.data.user.Session.getByUser(session, this).forEach(i -> i.delete(session));
```

Comment:
Use JPQL delete statement, to not fetch all entities. Example: https://github.com/cryptic-game/java-backend-dev/blob/master/java-data/src/main/java/net/cryptic_game/backend/data/user/Session.java#L104